packages/driver/src/cy/commands/navigation.coffee

```diff
@@ -564,6 +564,11 @@ module.exports = (Commands, Cypress, cy, state, config) ->
         ## for this, and so we need to resolve onLoad immediately
         ## and bypass the actual visit resolution stuff
         if bothUrlsMatchAndRemoteHasHash(current, remote)
+          if current.hash is remote.hash
+            consoleProps["Note"] = "Because this visit was to the same hash, the page did not reload and the onBeforeLoad and onLoad callbacks did not fire."
+
+            return Promise.resolve()
```

Comment:
This is the right approach to short circuiting and avoiding changing the `iframeSrc`, but the problem is that you're not calling the `onLoad` function, which is the thing that resolves with the remote `win` and ends up calling `onLoad`.

I know you don't want to call `options.onLoad` since it's already happened, so you'll need to take an argument that skips that check.

There are a few other things in the `onLoad` function that are necessary including setting `url` on the `options._log`, etc.